### PR TITLE
feat: be able to show viva completions in case card

### PR DESCRIPTION
**Pull request by karlknowit**
## Explain the changes you’ve made
Added the ability to display a completions list in the case card component in CaseOverview screen

Also did utilise the ApplicationStatusType in CaseOverview screen to avoid having dangling strings.

## Explain why these changes are made
As part of the task where a user should be able to see completions for its case, this PR is added. This PR adds the possibility for a user to see which completions is needed in caseoverview as a bullet list in CaseCard.

## Explain your solution
The CaseOverview screen checks the caseData property if any completions exists. If they exists then extract the descriptions from the from the complections object. 

## How to test
1. Checkout this branch
2. Fire upp the simulator by running the command `yarn ios`
3. Make sure you have a case with the status type `active:completionRequired:viva` or whichever type you want for testing purposes
4. Make sure you have a form (in formbuilder) with the BulletList question type
5. 
## Tested environments

- [] Storybook on a iOS device/simulator.
- [x] Building the Application on a iOS device/simulator.
- [] Building the Application on a Android device/simulator.

simulator_screenshot_7C24B8D1-54D9-49B1-8911-2907931F741E

### CODE REVIEW

**Comment by sebastianthulin** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 1 Issues 
* Code issue 1 Code issues 

 [Details](https://app.embold.io/organization/gh/helsingborg-stad/repositories/fedfce3fd8695f238dea9cb00bd37ce7/pull-request/602)

**Comment by helsingborgsstad** [comment]
Task linked: [CU-1pdtym9 Lista vilka uppgifter som behöver kompletteras i översiktsvyn](https://app.clickup.com/t/1pdtym9)

File: source/helpers/FormatCompletions.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    (previous: string[], current: CompletionsType) => {
+      const unapprovedCompletion = !current?.approved;
+
+      if (unapprovedCompletion) {
+        return [...previous, current.description];
+      }
+      return previous;
+    },
+    []
+  );
```
Comment (jlarsson) [inline source/helpers/FormatCompletions.ts:18]: A more concise approch is 
`completions.filter(({approved}) => approved).map(({description}) => description)`

**Review by jlarsson — CHANGES_REQUESTED** [review]
Favor filter+map over reduce

File: source/helpers/FormatCompletions.ts (hunk 1 above)
Comment (karlknowit) [reply source/helpers/FormatCompletions.ts:18]: I agree. We should use filter and map instead of reduce here, it makes the code more readable.

**Review by jlarsson — APPROVED** [review]

**Review by sepehrnoor — APPROVED** [review]
Looks good to me

**Review by MasterKenth — APPROVED** [review]
🤏

**Pull request merged**